### PR TITLE
[VirtualCluster] Fix `clusterversions` CRD, add `kind` support and review the demo flow

### PR DESCRIPTION
**Pull request by brightzheng100**
This PR introduces 3 major updates for VirtualCluster:

1. Fix the CRD as the generated one for `clusterversions` doesn't work
2. Add `kind` support, together with existing `minikube`
3. Review the demo flow with more comments for users to better understand VirtualCluster through the process

### CODE REVIEW

File: incubator/virtualcluster/doc/demo.md [hunk 1]
```diff
@@ -1,118 +1,256 @@
-# VirtualCluster Demo
+# VirtualCluster Walkthrough Demo
 
-This demo illustrates how to setup a virtualcluster in an existing `minikube` Kubernetes cluster.
+This demo illustrates how to setup a VirtualCluster in an existing lightweight environment, 
+be it [`minikube`](https://minikube.sigs.k8s.io/) or [`kind`](https://kind.sigs.k8s.io/docs/) Kubernetes cluster.
 
-All virtualcluster related API resources (CRD, Secret, Configmap etc.) are created in an
-"admin namespace" for the rest of this demo we'll use the `default` namespace.
+It should work exactly the same if you were working on any other Kubernetes distrubitions too.
```
Comment (christopherhein) [inline incubator/virtualcluster/doc/demo.md:6]:
```suggestion
It should work exactly the same if you were working on any other Kubernetes distributions too.
```

File: incubator/virtualcluster/doc/demo.md [hunk 2]
```diff
@@ -1,118 +1,256 @@
...
+
+For example, to spin up a `minukube` cluster:
```
Comment (christopherhein) [inline incubator/virtualcluster/doc/demo.md:8]:
```suggestion
For example, to spin up a `minikube` cluster:
```

File: incubator/virtualcluster/doc/demo.md [hunk 3]
```diff
@@ -1,118 +1,256 @@
...
-the following cmd.
+## (Optional) Create `kubelet` client secrete and update `vn-agent`
+
+By default, `vn-agent` works in a suboptimal mode by forwarding all `kubelet` API requests to super master.
+A more efficient method is to communicate with `kubelet` directly using the client cert/key used by the super master.
+
+The location of the client PKI files may vary based on the local setup.
+Please note that we need to make sure the client cert/key files are imported as `client.crt` and `client.key` so that they can be referenced to.
+
+### Create `kubelet` client secrete in `minikube` cluster
```
Comment (christopherhein) [inline incubator/virtualcluster/doc/demo.md:111]:
```suggestion
### Create `kubelet` client secret in `minikube` cluster
```

File: incubator/virtualcluster/doc/demo.md [hunk 4]
```diff
@@ -1,118 +1,256 @@
...
+
+
+## Access Virtual Cluster
+
+The generated `vc-1.kubeconfig` can be used as a normal `kubeconfig` to access the tenant virtual cluster.
+
+Please note that if you're working on `kind` cluster which, by default, exposes one random host port pointing to Kubernetes' default API Server port `6443`. In this case, we need to work around it and the simplest way is to deploy a "sidecar" container as the proxy to route management traffic to the service:
+
+```bash
+# Do this only when you're wroking in `kind`:
```
Comment (christopherhein) [inline incubator/virtualcluster/doc/demo.md:189]:
```suggestion
# Do this only when you're working in `kind`:
```

File: incubator/virtualcluster/doc/demo.md [hunk 5]
```diff
@@ -135,39 +273,59 @@ spec:
         command:
         - top
 EOF
-```
```
Comment (christopherhein) [inline incubator/virtualcluster/doc/demo.md:138]: Can you check the output is what you are looking for? it looks like it's all in a code block for the experiments section. Would it be easier to read for a user if the setups were broken by "normal" text between the code blocks?

**Review by christopherhein — CHANGES_REQUESTED** [review]
Nice work, a couple nits. Thanks for doing this @brightzheng100

File: incubator/virtualcluster/doc/demo.md (hunk 5 above)
Comment (brightzheng100) [reply incubator/virtualcluster/doc/demo.md:138]: Makes sense, one single code block is really too much.
Let me update that accordingly.

**Comment by zhuangqh** [comment]
please squash your commit

**Comment by Fei-Guo** [comment]
This looks great. Thanks a lot for working on this.

/lgtm
/approve

**Pull request merged**